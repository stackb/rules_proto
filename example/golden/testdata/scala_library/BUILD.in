# gazelle:proto_rule proto_compile implementation stackb:rules_proto:proto_compile

# gazelle:proto_plugin scala implementation scalapb:scalapb:protoc-gen-scala
# gazelle:proto_plugin scala option grpc
# gazelle:proto_plugin scala deps @com_google_protobuf//:protobuf_java
# gazelle:proto_plugin scala deps @maven_scala//:com_thesamet_scalapb_lenses_2_12
# gazelle:proto_plugin scala deps @maven_scala//:com_thesamet_scalapb_scalapb_runtime_2_12
# gazelle:proto_plugin scala deps @maven_scala//:com_thesamet_scalapb_scalapb_runtime_grpc_2_12
# gazelle:proto_plugin scala deps @maven_scala//:io_grpc_grpc_api
# gazelle:proto_plugin scala deps @maven_scala//:io_grpc_grpc_stub
# gazelle:proto_plugin scala deps @maven_scala//:io_grpc_grpc_protobuf

# gazelle:proto_plugin akka-grpc implementation akka:akka-grpc:protoc-gen-akka-grpc
# gazelle:proto_plugin akka-grpc deps @maven_akka//:com_lightbend_akka_grpc_akka_grpc_runtime_2_12
# gazelle:proto_plugin akka-grpc deps @maven_akka//:com_typesafe_akka_akka_actor_2_12
# gazelle:proto_plugin akka-grpc deps @maven_akka//:com_typesafe_akka_akka_http_core_2_12
# gazelle:proto_plugin akka-grpc deps @maven_akka//:com_typesafe_akka_akka_stream_2_12

# gazelle:proto_rule proto_scala_library implementation stackb:rules_proto:proto_scala_library
# gazelle:proto_rule proto_scala_library visibility //visibility:public

# gazelle:proto_rule grpc_scala_library implementation stackb:rules_proto:grpc_scala_library
# gazelle:proto_rule grpc_scala_library visibility //visibility:public

# gazelle:proto_language scala plugin scala
# gazelle:proto_language scala plugin akka-grpc
# gazelle:proto_language scala rule proto_compile
# gazelle:proto_language scala rule proto_scala_library
# gazelle:proto_language scala rule grpc_scala_library
