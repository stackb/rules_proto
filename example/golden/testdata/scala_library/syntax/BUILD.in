# gazelle:proto_rule proto_compile implementation stackb:rules_proto:proto_compile
# gazelle:proto_plugin scala implementation scalapb:scalapb:protoc-gen-scala
# gazelle:proto_plugin scala option grpc
# gazelle:proto_rule proto_scala_library implementation stackb:rules_proto:proto_scala_library
# gazelle:proto_rule proto_scala_library deps @build_stack_rules_proto//plugin/scalapb/scalapb:proto_scala_library_2_12
# gazelle:proto_rule proto_scala_library visibility //visibility:public
# gazelle:proto_language scala plugin scala
# gazelle:proto_language scala rule proto_compile
# gazelle:proto_language scala rule proto_scala_library
