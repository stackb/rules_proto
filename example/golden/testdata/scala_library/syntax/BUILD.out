load("@rules_proto//proto:defs.bzl", "proto_library")
load("@build_stack_rules_proto//rules/scala:proto_scala_library.bzl", "proto_scala_library")
load("@build_stack_rules_proto//rules:proto_compile.bzl", "proto_compile")

# gazelle:proto_rule proto_compile implementation stackb:rules_proto:proto_compile
# gazelle:proto_plugin scala implementation scalapb:scalapb:protoc-gen-scala
# gazelle:proto_plugin scala option grpc
# gazelle:proto_rule proto_scala_library implementation stackb:rules_proto:proto_scala_library
# gazelle:proto_rule proto_scala_library deps @build_stack_rules_proto//plugin/scalapb/scalapb:proto_scala_library_2_12
# gazelle:proto_rule proto_scala_library visibility //visibility:public
# gazelle:proto_language scala plugin scala
# gazelle:proto_language scala rule proto_compile
# gazelle:proto_language scala rule proto_scala_library

proto_library(
    name = "syntax_proto",
    srcs = [
        "noun.proto",
        "package.proto",
        "pronoun.proto",
    ],
    visibility = ["//visibility:public"],
    deps = ["//scalapb:scalapb_proto"],
)

proto_compile(
    name = "syntax_scala_compile",
    outputs = ["syntax_scala.srcjar"],
    plugins = ["@build_stack_rules_proto//plugin/scalapb/scalapb:protoc-gen-scala"],
    proto = "syntax_proto",
)

proto_scala_library(
    name = "syntax_scala_library",
    srcs = ["syntax_scala.srcjar"],
    visibility = ["//visibility:public"],
    deps = ["@build_stack_rules_proto//plugin/scalapb/scalapb:proto_scala_library_2_12"],
)
